Correcting major-mode for gerbil.el, see #510

doyougnu opened the pull request:
### What
This PR fixes #510, there were two problems:

1. `gerbil.el` did not set the `major-mode` to `gerbil-mode`
2. `gerbil.el` provided `gerbil` but spacemacs and `use-package` expected it to provide `gerbil-mode`

### The fix
1. rename `gerbil.el` to `gerbil-mode.el`
2. add `(setq major-mode 'gerbil-mode)`
3. change `(provide 'gerbil)` to `(provide 'gerbil-mode)`

This should not be an invasive change in any way, but I have not tested it in vanilla emacs so please try it and reject if it changes the workflow. If it impacts the vanilla emacs workflow in any way then I'll just pull a mirror copy for the `spacemacs` layer and maintain it. Any comments on ways to better contribute also appreciated.

Thanks!

fare commented:
Looks good to me. I would add however a note in `emacs.md` about steps to migrate configuration from `0.16` and earlier.

vyzo commented:
We also need to update the `install` script to install the correct file; it will try to copy `gerbil.el` now.

doyougnu commented:
I've fixed up the `install` script to copy `gerbil-mode`, noticed the same issue in the `emacs.md` and `use-package` documentation and also added a disclaimer to the bottom of the first section in `emacs.md` to refer to #510. 

In general I'm not a fan of linking to issues like that but I thought that was the wrong place to describe the migration.

vyzo commented on doc/guide/emacs.md:
```diff
@@ -29,6 +29,8 @@ You can then make `gxi` your scheme program by setting `scheme-program-name`:
 
 And you can now run Gerbil with `M-x run-scheme`.
 
+N.B. `gerbil-mode.el` had a breaking change as of version 0.16. See [here](https://github.com/vyzo/gerbil/issues/510) for migrating.
```
What we have to say is that "up to v0.16 the editing mode file was gerbil.el. Starting with `actual-version`, gerbil mode has been migrated to `gerbil-mode.el`; see [#510](...)"

vyzo replied:
Note that gerbil.el was shipped with v0.16 so the comment as is is factually incorrect.

doyougnu commented:
I've updated the relevant section of `emacs.md`. 

I want to be clear that I wasn't able to get a local build working on NixOS (with `nix-shell -p zlib openssl sqlite gambit --run './build.sh'`, and with @fare's `nixpkgs`) but I think this is an unrelated issue. So the version represented here is from @fare's `nixpkgs` branch and `gerbil-unstable`. I'm hesitant to merge the PR without testing it locally by performing the local build, but I think this is not the place for that discussion and could be a new issue?

vyzo commented on doc/guide/emacs.md:
```diff
@@ -29,6 +29,8 @@ You can then make `gxi` your scheme program by setting `scheme-program-name`:
 
 And you can now run Gerbil with `M-x run-scheme`.
 
+N.B. Up to v0.16 the editing mode file was gerbil.el. Starting with `v0.16-1-g36a31050`, gerbil mode has been migrated to gerbil-mode.el; see [#510](https://github.com/vyzo/gerbil/issues/510) for migrating.
```
```suggestion
N.B. Up to v0.16 the editing mode file was gerbil.el. After `v0.16-48-g46f10016`, gerbil mode has been migrated to gerbil-mode.el; see [#510](https://github.com/vyzo/gerbil/issues/510) for migrating.
```

vyzo replied:
Let's use the current master, it's `v0.16-48-g46f10016`
